.github/workflows/tests-e2e-upgrade.yaml

```diff
@@ -370,6 +370,50 @@ jobs:
       - name: Start conn-disrupt-test
         uses: ./.github/actions/conn-disrupt-test-setup
 
+      - name: Running pre-flight check
+        if: ${{ matrix.skip-upgrade != 'true' }}
+        shell: bash
+        run: |
+          # Running pre-flight check
+          helm template './untrusted/cilium-newest/install/kubernetes/cilium' \
```

Comment:
If you moved all of the internal logic for this preflight into a new CLI command - for example `cilium preflight` - then I think it would make it much easier for you to iterate on the functionality in order to test locally as well as to improve (a) the wait conditions and (b) how errors are reported when the preflight deployment fails.

Related: https://github.com/cilium/cilium/pull/38624#issuecomment-2776708686 .